### PR TITLE
documented  NODERAWFS

### PR DESCRIPTION
**Pull request by eleensmathew**
I wrote most from the notes available in https://github.com/emscripten-core/emscripten/blob/fde55d2f0f54f167a64a77ed013d2bbf22ab152d/src/settings.js#L881-L892.
Please check the pull request and give me feedback if there is something I should change

### CODE REVIEW

File: site/source/docs/api_reference/Filesystem-API.rst [hunk 1]
```diff
@@ -77,6 +77,17 @@ This file system lets a program in *node* map directories (via a mount operation
 
 See `this test <https://github.com/emscripten-core/emscripten/blob/main/tests/fs/test_nodefs_rw.c>`_ for an example.
 
+.. _filesystem-api-noderawfs:
+
+NODERAWFS
+------
```
Comment (kripken) [inline site/source/docs/api_reference/Filesystem-API.rst:83]:
```suggestion
---------
```
(to match the text above it)

File: site/source/docs/api_reference/Filesystem-API.rst [hunk 2]
```diff
@@ -77,6 +77,17 @@ This file system lets a program in *node* map directories (via a mount operation
...
+
+.. note:: This file system is only for use when running inside :term:`node.js`.
+
+ This is a special backend as it replaces all normal filesystem access with direct Node.js operations, without the need to do `FS.mount()`. The initial working directory will be same as process.cwd() instead of VFS root directory.  Because this mode directly uses Node.js to access the real local filesystem on your OS, the code will not necessarily be portable between OSes - it will be as portable as a Node.js program would be, which means that differences in how the underlying OS handles permissions and errors and so forth may be noticeable.  This has mostly been tested on Linux so far.
+
+See `this test <https://github.com/emscripten-core/emscripten/blob/main/tests/fs/test_noderawfs_nofollow.c>`_ for an example.
```
Comment (kripken) [inline site/source/docs/api_reference/Filesystem-API.rst:89]: Linking to the same test as above would be better, as it has a lot more detail there. Can also mention the `#ifndef NODERAWFS` block there, which shows the exact difference between NODEFS and NODERAWFS

**Review by eleensmathew — COMMENTED** [review]
I made the changes as you requested. If there are further changes to be made please tell me

File: site/source/docs/api_reference/Filesystem-API.rst (hunk 2 above)
Comment (kripken) [reply site/source/docs/api_reference/Filesystem-API.rst:100]: `#ifndef NODERAWFS` means that the code is *not* included in NODERAWFS mode. That is, the benefit of this mode is to not need to write that code. So we should clarify that in a comment, perhaps something like "Without NODERAWFS, you need to mount a directory for NODEFS. With NODERAWFS, that is not needed and you can simply access files immediately."

**Comment by kripken** [comment]
Also there is a CI error on building the docs,
```
Warning, treated as error:
/root/project/site/source/docs/api_reference/Filesystem-API.rst:5:Duplicate explicit target name: "this test".
Makefile:138: recipe for target 'text' failed
make: *** [text] Error 2
make: Leaving directory '/root/project/site'

Exited with code exit status 2
```
That will need to be fixed before landing too.

**Comment by eleensmathew** [comment]
Can you please check the documentation now? Also I don't know where the error is located. I keep changing the 'this test' part to see what's going wrong but I'm still getting the error.
"this test".
Makefile:138: recipe for target 'text' failed

**Comment by kripken** [comment]
Very good question. I don't understand that error either...

But I think it will go away if that entire line is deleted,
```rst
See `this test <https://github.com/emscripten-core/emscripten/blob/main/tests/fs/test_noderawfs_nofollow.c>`_ for an example.
```
And maybe we don't need that line. Also the code fragment might not be worth it. Just mentioning "See the above section on NODEFS, and in the link you can see a `mount` operation - that is not needed in NODERAWFS" is good enough I think.

**Comment by eleensmathew** [comment]
Ok, I cant see the error anymore when I removed the line. I also made the changes as you requested.

**Review by kripken — APPROVED** [review]
Thanks!

**Comment by eleensmathew** [comment]
Thanks for guiding me :)

**Pull request merged**